src/cluster/placement/algo/mirrored_test.go

```diff
@@ -380,73 +250,48 @@ func TestMirrorAddMultiplePairs(t *testing.T) {
 	assert.Equal(t, i7.Shards().AllIDs(), i8.Shards().AllIDs())
 	assert.Equal(t, i7.ShardSetID(), i8.ShardSetID())
 
-	// Removing all initializing nodes will trigger the revert path.
```

Comment:
This case here is still valid as well, right? As in if I did:

```
	p2, err := a.RemoveInstances(p1.Clone(), []string{"i5", "i6", "i7", "i8"})	
```

it would still hit the revert path right? I think we should probably add your case as a new test case while keeping this one.

Reply:
This one (revert all replaces) - yes, but the next one (partial revert) is now allowed while the test expected it would fail. Sure, makes sense to keep preserve this test with less modification and create a separate one.

Reply:
As pointed out in another comment, we should keep old behavior for add/remove operations.